subsys/metrics/src/main/java/org/commonjava/indy/metrics/conf/IndyMetricsConfig.java

```diff
@@ -0,0 +1,109 @@
+    public final static String INDY_METRICS_REPORTER_GRPHITEREPORTER_PORT = "GraphiteReporter.port";
+    public final static String INDY_METRICS_REPORTER_GRPHITEREPORTER_PREFIX = "GraphiteReporter.prefix";
+    public final static String INDY_METRICS_REPORTER_GRPHITEREPORTER_SIMPLE_PERIOD = "GraphiteReporter.simple.period";
+    public final static String INDY_METRICS_REPORTER_GRPHITEREPORTER_JVM_PERIOD = "GraphiteReporter.jvm.period";
+    public final static String INDY_METRICS_REPORTER_GRPHITEREPORTER_HEALTHCHECK_PERIOD = "GraphiteReporter.healthcheck.period";
+    public final static int INDY_METRICS_REPORTER_GRPHITEREPORTER_DEFAULT_PERIOD = 30;
+
+    private String reporter;
+    private String grphiterHostName;
+    private int grphiterPort;
+    private String grphiterPrefix ;
+    private int grphiterSimplePriod;
+    private int grphiterJVMPriod;
+    private int grphiterHealthcheckPeriod;
+
+    public String getReporter() {
+        return reporter;
+    }
+    @ConfigName(IndyMetricsConfig.INDY_METRICS_REPORTER)
+    public void setReporter(String reporter) {
+        this.reporter = reporter;
+    }
+
+    public String getGrphiterHostName() {
+        return grphiterHostName;
+    }
+
+    @ConfigName(IndyMetricsConfig.INDY_METRICS_REPORTER_GRPHITEREPORTER_HOSTNAME)
+    public void setGrphiterHostName(String grphiterHostName) {
+        this.grphiterHostName = grphiterHostName;
+    }
+
+    public int getGrphiterPort() {
+        return grphiterPort;
+    }
+
+    @ConfigName(IndyMetricsConfig.INDY_METRICS_REPORTER_GRPHITEREPORTER_PORT)
+    public void setGrphiterPort(int grphiterPort) {
+        this.grphiterPort = grphiterPort;
+    }
+
+    public String getGrphiterPrefix() {
+        return grphiterPrefix;
+    }
+
+    @ConfigName(IndyMetricsConfig.INDY_METRICS_REPORTER_GRPHITEREPORTER_PREFIX)
+    public void setGrphiterPrefix(String grphiterPrefix) {
+        this.grphiterPrefix = grphiterPrefix;
+    }
+
+    public int getGrphiterSimplePriod() {
+        return grphiterSimplePriod;
+    }
+
+    @ConfigName(IndyMetricsConfig.INDY_METRICS_REPORTER_GRPHITEREPORTER_SIMPLE_PERIOD)
+    public void setGrphiterSimplePriod(int grphiterSimplePriod) {
+        this.grphiterSimplePriod = grphiterSimplePriod;
+    }
+
+    public int getGrphiterJVMPriod() {
+        return grphiterJVMPriod;
+    }
+
+    @ConfigName(IndyMetricsConfig.INDY_METRICS_REPORTER_GRPHITEREPORTER_JVM_PERIOD)
+    public void setGrphiterJVMPriod(int grphiterJVMPriod) {
+        this.grphiterJVMPriod = grphiterJVMPriod;
+    }
+
+    public int getGrphiterHealthcheckPeriod() {
+        return grphiterHealthcheckPeriod;
+    }
+
+    @ConfigName(IndyMetricsConfig.INDY_METRICS_REPORTER_GRPHITEREPORTER_HEALTHCHECK_PERIOD)
+    public void setGrphiterHealthcheckPeriod(int grphiterHealthcheckPeriod) {
+        this.grphiterHealthcheckPeriod = grphiterHealthcheckPeriod;
+    }
+
+    public String getDefaultConfigFileName() {
```

Comment:
You can create a file in `src/main/conf/conf.d/metrics.conf`that contains the keys with their default values (or some example values if default values don't make sense...like for graphite hostname). Then, add comments in the file to tell what the configuration options control. If you have a set of config options like hostname, port, etc. all for one thing, like a graphite server, you can use a single comment section to document that set. Comments use the '#' character.

This allows us to bundle those sample configurations up and put them in the deployment tarball so it's easy for someone to come along and edit the config files, where the docs about the available options are right there.

You can see other src/main/conf/conf.d files in the addons/*/common modules for examples.